exporter/signalfxexporter/internal/dimensions/dimclient.go

```diff
@@ -32,7 +32,7 @@ import (
 // updates are currently not done by this port.
 type DimensionClient struct {
 	sync.RWMutex
-	ctx           context.Context
```

Comment:
Contexts aren't supposed to be stored, per [go's documentation](https://pkg.go.dev/context), so we should store the cancel instead.

Reply:
This is a part of the refactor as a whole, removing the stored context in favor of the cancel allows the dimclient to be able to start and stop by its own calls. If we leave it as is we're relying on the exporter's top level context cancel (which technically works), but then means we can't call `goleak` in this package and we're relying on another package to manage its lifecycle without being obvious.

I'd prefer to keep it as is, unless you think this is too much change (or too confusing) in one PR.